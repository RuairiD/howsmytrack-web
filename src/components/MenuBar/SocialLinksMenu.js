import React from "react";

import { Menu, Icon } from "antd";
import { A, Div, Span } from "lemon-reset";

const SocialLinksMenu = ({ isMobile, onMenuClick }) => {
    let style = {};
    if (!isMobile) {
        style = {
            position: "absolute",
            bottom: "0px",
        };
    }
    return (
        <Div style={style}>
            <Menu
                theme="light"
                onClick={onMenuClick}
                style={{
                    borderRight: 0,
                }}
            >
                {isMobile && <Menu.Divider />}
                <Menu.Item className="ant-menu-item">
                    <A href="https://twitter.com/howsmytrackcom" target="_blank" rel="noopener noreferrer" className="d-flex">
                        <Icon className="align-middle my-auto" type="twitter" />
                        <Span>Twitter</Span>
                    </A>
                </Menu.Item>
                <Menu.Item className="ant-menu-item">
                    <A href="https://discord.gg/6jGFP6N" target="_blank" rel="noopener noreferrer" className="d-flex">
                        <Icon className="align-middle my-auto" type="robot" />
                        <Span>Discord</Span>
                    </A>
                </Menu.Item>
            </Menu>
        </Div>
    );
};

export default SocialLinksMenu;
