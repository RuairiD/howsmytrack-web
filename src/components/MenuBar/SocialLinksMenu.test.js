import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import SocialLinksMenu from "./SocialLinksMenu";

describe("SocialLinksMenu", () => {
    it("renders Twitter and Discord links within a Menu", () => {
        const wrapper = shallow(
            <SocialLinksMenu
                onMenuClick={() => {}}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders without absolute positioning and includes an additional Divider for mobile clients", () => {
        const wrapper = shallow(
            <SocialLinksMenu
                isMobile
                onMenuClick={() => {}}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
