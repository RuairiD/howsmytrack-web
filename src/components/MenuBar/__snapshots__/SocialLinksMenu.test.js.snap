// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLinksMenu renders Twitter and Discord links within a Menu 1`] = `
<Div
  className=""
  style={
    Object {
      "bottom": "0px",
      "position": "absolute",
    }
  }
>
  <Menu
    onClick={[Function]}
    style={
      Object {
        "borderRight": 0,
      }
    }
    theme="light"
  >
    <MenuItem
      className="ant-menu-item"
    >
      <A
        className=""
        href="https://twitter.com/howsmytrackcom"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Icon
          type="twitter"
        />
        <Span
          className=""
        >
          Twitter
        </Span>
      </A>
    </MenuItem>
    <MenuItem
      className="ant-menu-item"
    >
      <A
        className=""
        href="https://discord.gg/6jGFP6N"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Icon
          type="robot"
        />
        <Span
          className=""
        >
          Discord
        </Span>
      </A>
    </MenuItem>
  </Menu>
</Div>
`;

exports[`SocialLinksMenu renders without absolute positioning and includes an additional Divider for mobile clients 1`] = `
<Div
  className=""
  style={Object {}}
>
  <Menu
    onClick={[Function]}
    style={
      Object {
        "borderRight": 0,
      }
    }
    theme="light"
  >
    <Divider
      className=""
      disabled={true}
      style={Object {}}
    />
    <MenuItem
      className="ant-menu-item"
    >
      <A
        className=""
        href="https://twitter.com/howsmytrackcom"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Icon
          type="twitter"
        />
        <Span
          className=""
        >
          Twitter
        </Span>
      </A>
    </MenuItem>
    <MenuItem
      className="ant-menu-item"
    >
      <A
        className=""
        href="https://discord.gg/6jGFP6N"
        rel="noopener noreferrer"
        target="_blank"
      >
        <Icon
          type="robot"
        />
        <Span
          className=""
        >
          Discord
        </Span>
      </A>
    </MenuItem>
  </Menu>
</Div>
`;
